test/ecto/type/enum_test.exs

```diff
@@ -0,0 +1,98 @@
+defmodule Ecto.Type.EnumTest do
```

Comment:
Those tests are good but they are too high-level. We also need to unit test parameterized types directly through the Ecto.Type API.